### ISSUE TITLE
Concatenation not working

### ISSUE THREAD

**Issue by apuatcfbd** [toddbluhm/env-cmd#358]
```
REACT_APP_URL_PROTOCOL=https
REACT_APP_URL=site.com

REACT_APP_BASE_URL="${REACT_APP_URL_PROTOCOL}://${REACT_APP_URL}"
```
This format was working with CRA but this is not working with `env-cmd`,

used like `"build:test": "env-cmd -f .env.production.test react-app-rewired build"`

Is there any way to make this work? Concatenation is important for my case

**Comment by pjetrucha** [comment]
@apuatcfbd 
As a workaround, put a variable with concatenation (REACT_APP_BASE_URL) in env files that are default for CRA, such as:
https://create-react-app.dev/docs/adding-custom-environment-variables/#what-other-env-files-can-be-used

**Comment by apuatcfbd** [comment]
Thanks @pjetrucha but I didn't get it. I've different env's for different stages. in that scenario how do I do that?
I need this (for same env's vars): https://create-react-app.dev/docs/adding-custom-environment-variables/#expanding-environment-variables-in-env

**Comment by pjetrucha** [comment]
The CRA with env-cmd merges all the files together, so in your example, it will search for:

`.env.production.test`, `.env.production.local`, `.env.local`, `.env.production`, `.env`

The first file is from env-cmd, so variable expansion doesn't work in it, but you can move it to one of 4 other files that CRA will look for, e.g.:

```env
# .env.production.test
REACT_APP_URL_PROTOCOL=https
REACT_APP_URL=site.com

# .env
REACT_APP_BASE_URL="${REACT_APP_URL_PROTOCOL}://${REACT_APP_URL}"
```

**Comment by toddbluhm** [comment]
Yeah, this needs to be looked into/fixed...possibly by one of the PRs attached 

@apuatcfbd is what you have listed in your post the exact reproduction case, or is there additional data needed? In other words, what is the simplest use case so that I can recreate a test case for it?

**Comment by apuatcfbd** [comment]
For example, I need a base domain like `example.com` then I've other domains like `api.example.com` ...
In this case, I want to use concatenation for not having a duplicated domain name.
